code/game/objects/items/stacks/medical.dm

```diff
@@ -134,18 +166,152 @@
+	heal_brute = 15
+	grind_results = list(/datum/reagent/medicine/polypyr = 1)
+
+/obj/item/stack/medical/mesh
+	name = "regenerative mesh"
+	desc = "A bacteriostatic mesh used to dress burns."
+	gender = PLURAL
+	singular_name = "regenerative mesh"
+	icon_state = "regen_mesh"
+	self_delay = 3 SECONDS
+	other_delay = 1 SECONDS
+	amount = 15
+	max_amount = 15
+	repeating = TRUE
+	heal_burn = 10
+	var/is_open = TRUE ///This var determines if the sterile packaging of the mesh has been opened.
+	grind_results = list(/datum/reagent/medicine/spaceacillin = 2)
+
+/obj/item/stack/medical/mesh/Initialize()
+	..()
+	if(amount == max_amount)	 //only seal full mesh packs
+		is_open = FALSE
+		update_icon()
+
+
+/obj/item/stack/medical/mesh/update_icon()
+	if(is_open)
+		return ..()
+	icon_state = "regen_mesh_closed"
+
+/obj/item/stack/medical/mesh/try_heal(mob/living/M, mob/user, silent = FALSE)
+	if(!is_open)
+		to_chat(user, "<span class='warning'>You need to open [src] first.</span>")
+		return
+	return ..()
+
+/obj/item/stack/medical/mesh/AltClick(mob/living/user)
+	if(!is_open)
+		to_chat(user, "<span class='warning'>You need to open [src] first.</span>")
+		return
+	return ..()
+
+/obj/item/stack/medical/mesh/attack_hand(mob/user)
+	if(!is_open & user.get_inactive_held_item() == src)
+		to_chat(user, "<span class='warning'>You need to open [src] first.</span>")
+		return
+	return ..()
+
+/obj/item/stack/medical/mesh/attack_self(mob/user)
+	if(!is_open)
+		is_open = TRUE
+		to_chat(user, "<span class='notice'>You open the sterile mesh package.</span>")
+		update_icon()
+		playsound(src, 'sound/items/poster_ripped.ogg', 20, TRUE)
+		return
+	. = ..()
+
+/obj/item/stack/medical/mesh/advanced
+	name = "advanced regenerative mesh"
+	desc = "An advanced mesh made with aloe extracts and sterilizing chemicals, used to treat burns."
+
+	gender = PLURAL
+	icon_state = "aloe_mesh"
+	heal_burn = 15
+	grind_results = list(/datum/reagent/consumable/aloejuice = 1)
+	merge_type = /obj/item/stack/medical/mesh/advanced
+
+/obj/item/stack/medical/mesh/advanced/update_icon()
+	if(is_open)
+		return ..()
+	icon_state = "aloe_mesh_closed"
+
+/obj/item/stack/medical/aloe
+	name = "aloe cream"
+	desc = "A healing paste for minor cuts and burns."
+
+	gender = PLURAL
+	singular_name = "aloe cream"
+	icon_state = "aloe_paste"
+	self_delay = 2 SECONDS
+	other_delay = 1 SECONDS
+	novariants = TRUE
+	amount = 20
+	max_amount = 20
+	repeating = TRUE
+	heal_brute = 3
+	heal_burn = 3
+	grind_results = list(/datum/reagent/consumable/aloejuice = 1)
+	merge_type = /obj/item/stack/medical/aloe
+
+/obj/item/stack/medical/poultice
+	name = "mourning poultices"
+	singular_name = "mourning poultice"
+	desc = "A type of primitive herbal poultice.\nWhile traditionally used to prepare corpses for the mourning feast, it can also treat scrapes and burns on the living, however, it is liable to cause shortness of breath when employed in this manner.\nIt is imbued with ancient wisdom."
```

Comment:
```suggestion
	desc = "A type of primitive herbal poultice.\nWhile traditionally used to prepare corpses for the mourning feast, it can also treat scrapes and burns on the living. However, it is liable to cause shortness of breath when employed in this manner.\nIt is imbued with ancient wisdom."
```